frontend/app/src/components/profitloss/ReportsTable.vue

```diff
@@ -0,0 +1,189 @@
+
+<script lang="ts">
+import { PagedResourceParameters } from '@rotki/common';
+import { PagedReport } from '@rotki/common/lib/reports';
+import {
+  computed,
+  defineComponent,
+  onBeforeMount,
+  Ref,
+  ref
+} from '@vue/composition-api';
+import { DataTableHeader } from 'vuetify';
+import DateDisplay from '@/components/display/DateDisplay.vue';
+import DataTable from '@/components/helper/DataTable.vue';
+import CardTitle from '@/components/typography/CardTitle.vue';
+import { setupStatusChecking } from '@/composables/common';
+import i18n from '@/i18n';
+import { Section } from '@/store/const';
+import { ReportActions } from '@/store/reports/const';
+import { RotkehlchenState } from '@/store/types';
+import { useStore } from '@/store/utils';
+
+export default defineComponent({
+  name: 'ReportsTable',
+  components: {
+    DataTable,
+    CardTitle,
+    DateDisplay
+  },
+  setup() {
+    const selected: Ref<string[]> = ref([]);
+    const store = useStore();
+
+    const state: RotkehlchenState = store.state;
+    const itemsPerPage = state.settings!!.itemsPerPage;
+
+    const payload = ref<PagedResourceParameters>({
+      limit: itemsPerPage,
+      offset: 0,
+      orderByAttribute: 'timestamp',
+      ascending: false
+    });
+
+    const entries = computed(() => {
+      const state: RotkehlchenState = store.state;
+      return state.reports!!.reports;
+    });
+    const limit = computed(() => {
+      const state: RotkehlchenState = store.state;
+      return state.reports!!.reportsLimit;
+    });
+    const found = computed(() => {
+      const state: RotkehlchenState = store.state;
+      return state.reports!!.reportsFound;
+    });
+
+    const fetchReports = async (refresh: boolean = false) => {
+      await store.dispatch(`reports/${ReportActions.FETCH_REPORTS}`, {
+        ...payload.value,
+        onlyCache: !refresh
+      });
+    };
+    const loadReport = async (reportId: number, refresh: boolean = false) => {
+      const state: RotkehlchenState = store.state;
+      state.reports!!.reportId = reportId;
+      await store.dispatch(`reports/${ReportActions.FETCH_REPORT}`, {
+        ...payload.value,
+        onlyCache: !refresh
+      });
+    };
+    const deleteReport = async (reportId: number) => {
+      const state: RotkehlchenState = store.state;
+      state.reports!!.reportId = reportId;
+      await store.dispatch(`reports/${ReportActions.DELETE_REPORT}`, {});
+    };
+    const refresh = async () => await fetchReports(true);
+    const onPaginationUpdate = ({
+      ascending,
+      page,
+      sortBy,
+      itemsPerPage
+    }: {
+      page: number;
+      itemsPerPage: number;
+      sortBy: keyof PagedReport;
+      ascending: boolean;
+    }) => {
+      const offset = (page - 1) * itemsPerPage;
+      payload.value = {
+        ...payload.value,
+        orderByAttribute: sortBy,
+        offset,
+        limit: itemsPerPage,
+        ascending
+      };
+      fetchReports().then();
```

Comment:
@LefterisJP @kelsos I'm pretty sure for paging of reports `fetchReports` needs an argument of the above payload, and this needs to be handled below in `fetchReports` by adding it to the axios config key 'params' with the payload argument as a value.  The ProfitLossEvents need to refactor according to ReportsTable to issue similar paging on an API service like `fetchReportData` when using that component's data table as well.